### DISCUSSION TITLE
How to mark the required mandatory option with a small red star in the input component

### DISCUSSION THREAD

**Discussion by xiaozhiagi** [zauberzeug/nicegui#1395] [Q&A]
### Question

How to mark the required mandatory option with a small red star in the input component。For example, displaying a red small red star in front of a mandatory option to remind the user that it is a required item, while not indicating a small red star indicates that it is an optional option

**Comment by Anindya088** [comment]
Hi @txhy2018 ,

I suppose what you meant is how to make an input field required/mandatory in a form, roughly speaking. 
I think the red star is just a label, that itself will not make it mandatory, but you can put some logic checks in the submit button. I have taken an attempt to do this, probably there are better ways to do this, I am still learning the ropes :)

Please let me know if this is what you were looking for, or if I did not get the question correct.

```python
with ui.row().classes('gap-0'):
    ui.icon('emergency').classes('text-negative text-[10px] pr-1')
    ui.label('Username')
username = ui.input().props('dense outlined')
err_username = ui.label('Username is mandatory').classes('text-negative text-xs font-bold')
err_username.visible = False
ui.separator()
ui.label('Email')
ui.input().props('dense outlined')

ui.button('Submit', on_click=lambda: check_username())

def check_username():
    if not username.value:
        err_username.visible = True
    else:
        err_username.visible = False
        # Your next steps
```

The output will look something like this:
![image](https://github.com/zauberzeug/nicegui/assets/65127974/483dc1f8-5e63-44ea-9f1b-0a8b83c92ee9)


But the star itself does nothing, it's basically the logic check associated with the submit button. So in this case, if you keep it empty it will display the error message like this
![image](https://github.com/zauberzeug/nicegui/assets/65127974/d9183c13-7d91-4216-aa33-be561abe9206)

You can probably make it even more interesting, like the submit button itself is disabled until some values are input into username, etc. But the main idea will be this. 


Thanks,
Anindya
